toolkit/tools/pkg/osmodifierlib/modifierutils.go

```diff
@@ -21,5 +28,100 @@ func doModifications(baseConfigPath string, osConfig *imagecustomizerapi.OS) err
 		return err
 	}
 
+	if osConfig.Overlays != nil {
+		bootCustomizer, err := imagecustomizerlib.NewBootCustomizer(dummyChroot)
```

Comment:
The `bootCustomizer` variable here is hiding the `bootCustomizer` in the parent scope because you are using `:=` instead of `=`. So the `if bootCustomizer == nil` check below doesn't actually do anything since the parent's scope `bootCustomizer` variable will always be `nil` at that point.